### ISSUE TITLE
Weird type behavior in parallel comprehension

### ISSUE THREAD

**Issue by malmaud** [IntelLabs/ParallelAccelerator.jl#7]
```julia
julia> @acc f(x)=Float64[_ for _ in x]
julia> f([1.5,2.])
2-element Array{Int64,1}:
 1
 2
julia> versioninfo()
Julia Version 0.5.0-dev+887
Commit 1079755 (2015-10-21 16:21 UTC)
Platform Info:
  System: Darwin (x86_64-apple-darwin14.5.0)
  CPU: Intel(R) Core(TM) i7-3720QM CPU @ 2.60GHz
  WORD_SIZE: 64
  BLAS: libopenblas (USE64BITINT DYNAMIC_ARCH NO_AFFINITY Sandybridge)
  LAPACK: libopenblas64_
  LIBM: libopenlibm
  LLVM: libLLVM-3.3
```

**Comment by ehsantn** [comment]
We don't support 0.5 yet. Could you reproduce this on 0.4?

**Comment by ehsantn** [comment]
We have had issues with return types before. @ninegua Paul, what do you think?

**Comment by ninegua** [comment]
It was indeed a bug, and should be fixed by da47e3ff3b2015a9c013696e3177a832e8285009.

**Comment by ehsantn** [comment]
Thanks, Paul. @malmaud Thank you for reporting the bug.
